cdap-security/src/main/java/io/cdap/cdap/security/authorization/AuthorizerInstantiator.java

```diff
@@ -281,33 +285,210 @@ private void ensureValidAuthExtensionJar(File authorizerExtensionJar) throws Inv
 
   @Override
   public void close() throws IOException {
+    Authorizer authorizer = this.authorizer;
+
     if (authorizer != null) {
+      ClassLoader authorizerClassLoader = authorizer.getClass().getClassLoader();
+      ClassLoader oldCL = ClassLoaders.setContextClassLoader(authorizerClassLoader);
       try {
         authorizer.destroy();
       } catch (Throwable t) {
         LOG.warn("Failed to destroy authorizer.", t);
+      } finally {
+        if (authorizerClassLoader instanceof Closeable) {
```

Comment:
Sorry if it's obvious, but are we sure that this classloader is always the one we loaded before? If it possible that authorize class is actually loaded from the parent classloader?

Reply:
Updated